src/plugins/formActions/CreateAction.js

```diff
@@ -50,19 +51,15 @@ export default class CreateAction extends PropertiesAction {
                 return;
             }
 
-            const properties = key.split('.');
-            let object = this.domainObject;
-            const propertiesLength = properties.length;
-            properties.forEach((property, index) => {
-                const isComplexProperty = propertiesLength > 1 && index !== propertiesLength - 1;
-                if (isComplexProperty && object[property] !== null) {
-                    object = object[property];
-                } else {
-                    object[property] = value;
-                }
-            });
-
-            object = value;
+            const existingValue = this.domainObject[`${key}`];
+            if (!(existingValue instanceof Array) && (typeof existingValue === 'object')) {
+                value = {
+                    ...existingValue,
+                    ...value
+                };
+            }
+
+            _.set(this.domainObject, `${key}`, value);
```

Comment:
These feel like they should mutations inside a transaction.

Reply:
This is just setting properties on an object. Not calling objects.mutation.

Reply:
Right, but all object mutation should go through the mutation API. I suspect it's not here in order to avoid triggering persistence, but the solution to that is to create a transaction to queue up a bunch of changes, then persist them.

Reply:
Done